jib-core/src/main/java/com/google/cloud/tools/jib/cache/DefaultCacheStorageWriter.java

```diff
@@ -101,6 +110,36 @@ private static DescriptorDigest getDiffIdByDecompressingFile(Path compressedFile
     }
   }
 
+  /**
+   * Writes a json template to the destination path by writing to a temporary file then moving the
+   * file.
+   *
+   * @param jsonTemplate the json template
+   * @param destination the destination path
+   * @throws IOException if an I/O exception occurs
+   */
+  private static void writeMetadata(JsonTemplate jsonTemplate, Path destination)
+      throws IOException {
+    try (LockFile ignored1 = LockFile.lock(Paths.get(destination.toString() + ".lock"))) {
+      Path temporaryFile = Files.createTempFile(destination.getParent(), null, null);
+      temporaryFile.toFile().deleteOnExit();
+      Blobs.writeToFileWithLock(JsonTemplateMapper.toBlob(jsonTemplate), temporaryFile);
```

Comment:
Since we don't need a lock for the temp file, I'd like to avoid the "write-with-lock" methods. It may confuse people who read this code later.